src/bout++.cxx

```diff
@@ -221,7 +222,16 @@ int BoutInitialise(int &argc, char **&argv) {
   if (output.open("%s/BOUT.log.%d", data_dir, MYPE)) {
     return 1;
   }
-
+  {
+    const size_t len=512;
+    char buf[len];
+    snprintf(buf,len,"%s/.BOUT.pid.%d",data_dir, MYPE);
+    FILE * fp = fopen(buf, "w");
+    if (fp == nullptr)
+      return 1;
+    fprintf(fp,"%d\n",getpid());
+    fclose(fp);
+  }
```

Comment:
Could this be rewritten as idiomatic C++ rather than C-style?
Something like

```cpp
std::string filename;
std::stringstream(filename) >> data_dir >> "/.BOUT.pid." >> MYPE;
std::ofstream pid_file;
pid_file.open(filename, std::ios::out);
if (pid_file.is_open()) {
    pid_file << getpid() << "\n";
    pid_file.close()
}
```